File: 3-Problem04.cpp
/* NIM/NAMA	: 18216011/Akbar Ghifari
 * TANGGAL	: 21/05/18
 * SOAL		: BAB 3 PROBLEM 4 (Bilangan Prima)
 */

#include <iostream>
using namespace std;

bool prime (int X){
    if ((X==1)||(X==0)){
        return false;
    } else {
        int divisor = X-1;
        int counter = 0;
        while (divisor != 1){
            if (X % divisor == 0){
                counter++;
            }
            divisor--;
        }
        //EKSKULSIF 1 DAN X
        if (counter == 0){
            return true;
        } else {
            return false;
        }
    }
}

int main(){
    int A;
    int B;

    cout << "Masukkan nilai A : ";
    cin >> A;
    cout << "Masukkan nilai B : ";
    cin >> B;

    if (B < A){
        int temp;
        temp = A;
        A = B;
        B = temp;
    }

    cout << "Bilangan prima diantara " << A << " sampai " << B << " adalah:" << endl;
    while (A != B+1){
        if (prime(A)){
            cout << A << endl;
        }
        A++;
    }
    return 0;
}

File: 3-Problem04.pas
program primenumbers;
var
	i, A, B : integer;
	// A : batasan awal range, B : batasan akhir range

function isPrime(x : integer) : boolean;
// Fungsi isPrime akan mengembalikan true jika x merupakan bilangan prima
var
	j : integer;
begin
	isPrime := true;
	for j:=2 to x div 2 do
	begin
		if (x mod j) = 0 then
		begin
			isPrime := false;
		end;
	end;
end;

begin
	write('Masukkan nilai A : ');
	readln(A);
	write('Masukkan nilai B : ');
	readln(B);
	writeln('Bilangan prima diantara ',A, ' sampai ',B, ' adalah:');
	if (A < B) then
	begin
		for i:=A to B do
		begin
			if (isPrime(i) = true and (i <> 0) and (i <> 1)) then
			begin
				writeln(i); // mencetak ke layar jika angka tsb merupakan bilangan prima
			end;
		end;
	end;
	if (B < A) then
	begin
		for i:=B to A do
		begin
			if (isPrime(i) = true and (i <> 0) and (i <> 1)) then
			begin
				writeln(i); // mencetak ke layar jika angka tsb merupakan bilangan prima
			end;
		end;
	end;
end.